### PR TITLE
Incremental output for LLM entrypoint

### PR DESCRIPTION
**Pull request by yhu422**
Added an option to stream the outputs of each engine step, instead of waiting until all requests are finished.

### CODE REVIEW

**Review by simon-mo — COMMENTED** [review]
Can you create an example script in examples directory?

**Comment by robertgshaw2-neuralmagic** [comment]
cc @mgoin

**Comment by mgoin** [comment]
Yay this would be very convenient, thank you for implementing!

File: examples/offline_inference_stream.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+index = 0
+for output in outputs:
+    prompt = output.prompt
+    generated_text = output.outputs[0].text
+    stripped_text = generated_text[index:]
+    completion += stripped_text
+    print(f"{stripped_text!r}")
+    index = len(generated_text)
+#print complete response by assembling results from each iteration.
+print(f"Prompt: {prompts[0]!r}, Generated Text: {completion!r}")
```
Comment (rkooo567) [inline examples/offline_inference_stream.py:27]: newline!

File: vllm/entrypoints/llm.py [hunk 2]
```diff
@@ -187,6 +189,12 @@ def _add_request(
                                     prompt_token_ids,
                                     lora_request=lora_request)
 
+    def _stream_engine(self) -> Generator[RequestOutput, any, None]:
+        while self.llm_engine.has_unfinished_requests():
```
Comment (rkooo567) [inline vllm/entrypoints/llm.py:193]: assert use_tqdm is False, ("tqdm is not supported when stream output is used")

**Review by rkooo567 — COMMENTED** [review]
We should add a test btw. maybe we can add a test to tests/engine and new file such as test_stream_engine.py?

**Review by rkooo567 — COMMENTED** [review]
Btw, it may have problems if you send 2 generate API? 


```python3
outputs = llm.generate(prompts, sampling_params, stream=True)
outputs_2 = llm.generate(prompts2, sampling_params, stream=True))
```

In this case, wouldn't the first `outputs` mix up the result from `outputs_2`? How do we isolate responses in this case?

**Comment by WoosukKwon** [comment]
Why don't we add another API for this instead of expanding `LLM`?

**Comment by simon-mo** [comment]
I think the original idea is openai style API also have `stream` flag that change the behavior of the output

**Pull request closed without merging**